docs/src/checklists.md

```diff
@@ -0,0 +1,38 @@
+# Checklists
+
+The purpose of this page is to collate a series of checklists for commonly
+performed changes to the source code of Documenter.
+
+In each case, copy the checklist into the description of the pull request.
+
+## Making a release
+
+In preparation for a release, use the following checklist.
+
+````
+## Pre-release
+
```

Comment:
We have only really used release branches when we had breaking(-ish) changes on master, and we wanted to tag some changes without a 0.x minor version bump, just a patch bump. Since we don't need to worry about that anymore, I don't think we'll be using release branches (unless some need comes up where we have to "LTS" some branch).

Reply:
Ok, makes sense… Case 1 is what I meant by "release branch" (not a long-running branch for backports or anything like that.